packages/flutter_tools/lib/src/dart/analysis.dart

```diff
@@ -46,7 +46,9 @@ class AnalysisDriver {
 
   DriverOptions options;
 
-  String get sdkDir => options.dartSdkPath ?? cli_util.getSdkDir().path;
+  String get sdkDir {
+    return options.dartSdkPath ?? path.absolute(cli_util.getSdkDir().path);
```

Comment:
So this is the only command which respects dartSdkPath?  Odd that format above doesn't for example.